src/auth/models/strategy.class.ts

```diff
@@ -3,15 +3,24 @@ import * as express from 'express'
 import { NextFunction, Request, RequestHandler, Response } from 'express'
 import passport from 'passport'
 import { AUTH } from '../auth.constants'
+import { OAuth2Metadata } from '../oauth2'
+import { OpenIDMetadata } from '../oidc'
+import { FileSessionMetadata, RedisSessionMetadata } from '../session/models/sessionMetadata.interface'
 
-export class Authentication extends events.EventEmitter {
-    protected readonly strategyName: string
+export abstract class Strategy extends events.EventEmitter {
+    public readonly strategyName: string
     protected readonly router = express.Router({ mergeParams: true })
     constructor(strategyName: string) {
         super()
         this.strategyName = strategyName
     }
 
+    public abstract configure(
+        options: OAuth2Metadata | OpenIDMetadata | FileSessionMetadata | RedisSessionMetadata,
```

Comment:
why does the base authentication class have session stuff? sorry not seen it before. but session should live in it's own layer/folder etc, it's independent of authentication